.github/workflows/main.yml

```diff
@@ -104,7 +104,7 @@ jobs:
       fail-fast: true
       matrix:
         dotnetversion: [3.1.301]
-        goversion: [1.17.x]
```

Comment:
I needed to update our Go deps because I needed to update our codegen version to use enums.

So I did all that in this PR for simplicity. Happy to break out into a separate PR if that's preferable